### ISSUE TITLE
[MD077] Use MkDocs indentation even in Commonmark documents

### ISSUE THREAD

**Issue by skyzyx** [rvben/rumdl#784]
### Problem or motivation

I use `rumdl` alongside [`editorconfig-checker`](https://editorconfig-checker.github.io), which is a tool that enforces that documents follow the [`.editorconfig`](https://editorconfig.org) configuration.

When applied to `.md` documents with `indent = 2`, this means that second lines must be indented 4 spaces, then 6, etc. My documents are GFM documents, and to pass the `editorconfig-checker` check, next lines must be 4 spaces. However `MD077` instructs Commonmark documents to use 3 spaces. If it were an MkDocs document, it could be 4. But there does not seem to be a way to instruct `rumdl` to use 4 spaces intentionally.

This causes a conflict during the linting/fixing stage of my development pipeline with tools arguing with each other.

### Proposed solution

* Let `.editorconfig` override what MD077 validates against.
* Allow Commonmark documents to use MkDocs indentation rules.
* Allow the user to somehow specify $2n$, $3n$, $4n$, …, etc., where $n$ is the indentation level.

### Alternatives considered

Disable MD077.

### Feature category

Improve existing rule

### Example markdown

This is what is expected by `rumdl` with 2-space indentation and a second line for a Commonmark document.

```markdown
* This is a sentence in a list item.

   This is a second sentence in a list item.

```

This is what I'd like to achieve as a valid configuration.

```markdown
* This is a sentence in a list item.

    This is a second sentence in a list item.
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

### Additional context

I haven't learned Rust yet. Sorry.

**Comment by rvben** [comment]
Implemented in v0.2.50, as an `indent` option on MD077:

```toml
[MD077]
indent = 4
```

With that set, the document you asked for is clean, and `rumdl fmt` produces it from the 3-space form:

```console
$ rumdl check want.md
Success: No issues found in 1 file
```

The requirement is per item: the item's own marker column plus the configured indent. So a top-level item needs 4, and a nested item whose marker sits at column 2 needs 6, which is the shape you described. `indent = 2` works the same way if you prefer the tighter form.

That also removes the collision you hit. Without the option the requirement follows each item's content column, which is 3 for `1. ` and 2 for `* `, so a document mixing the two produces both an odd and an even continuation and `editorconfig-checker` can only accept one of them. A configured indent makes every item agree.

On the `.editorconfig` route: rumdl does read `.editorconfig` when you set `[global] editorconfig = true`, but `indent_size` maps onto MD007's list indent, not MD077's continuation indent, so it would not have answered this. Even if it did, your `indent_size = 2` would have chosen a 2-space continuation rather than the 4-space form in your example. The explicit key is what lets you say which one you want.

Thanks to @DavidPandleton, who implemented this in #786.

**Comment by skyzyx** [comment]
Thank you!
